src/kontrolnaya2/list.c

```diff
@@ -0,0 +1,71 @@
+#include <stdio.h>
+#include "list.h"
+
+
+Node* insertValue(Node *p, int value)
+{
+    Node *t = malloc(sizeof(Node));
+    t->value = value;
+    t->next = p;
+    t->previous = NULL;
+    p = t;
+    if (t->next != NULL) {
+        t->next->previous = t;
+    }
+    return p;
+}
+
+void symmetry(Node *top, Node *end)
+{
+    Node *counter = top;
+    Node *topTemporary = top;
+    Node *endTemporary = end;
+    int count = 0;
+    while (counter->next != NULL) {
+        count++;
+        counter = counter->next;
+    }
+    count++;
+
+    int flag = 1;
+    for (int i = 0; (count+1) / 2 >= i; i++) {
```

Comment:
Первый раз вижу, чтобы условие цикла писали в таком порядке

```suggestion
    for (int i = 0; i <= (count+1) / 2; i++) {
```